src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -801,6 +824,23 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
 
                             File.WriteAllText(tempHashPath, packageHash);
 
+                            // get hash for the unsigned content of signed package
+                            var newHash = packageDownloader.SignedPackageReader.GetContentHashForSignedPackage(cancellationToken);
```

Comment:
Since this code is exactly the same as line 517-533, should it be in a private function?